src/sentry_plugins/github/endpoints/webhook.py

```diff
@@ -33,11 +33,14 @@ class PushEventWebhook(Webhook):
     def __call__(self, organization, event):
         authors = {}
 
-        repo = Repository.objects.get_or_create(
-            organization_id=organization.id,
-            provider='github',
-            external_id=event['repository']['full_name'],
-        )[0]
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='github',
+                external_id=six.text_type(event['repository']['id']),
+            )
+        except Repository.DoesNotExist:
+            return 404
```

Comment:
tbh you can just do `raise Http404` like in normal Django and this should just work, without awkwardly returning just a status code.

If you wanted to change the API, I'd suggest we make the handlers return an `HttpResponse` object as a whole.